src/components/typeahead/typeahead.vue

```diff
@@ -0,0 +1,261 @@
+<script>
+    const UID_key = '__pd_ta_uid'
+
+    /* Implementation of lodash.get function */
+    function getProp( object, keys, defaultVal ){
+        keys = Array.isArray( keys )? keys : keys.split('.');
+        object = object[keys[0]];
+        if( object && keys.length>1 ){
+            return getProp( object, keys.slice(1) );
+        }
+        return object === undefined? defaultVal : object;
+    }
+
+    export default {
+        name: 'pd-typeahead',
+        props: {
+            items: {
+                required: true,
+                type: Array,
+            },
+            startAt: {
+                type: Number,
+                default: 3
+            },
+            multiSelect: {
+                type: Boolean,
+                default: false
+            },
+            filterKeys: {
+                type: Array,
+                required: true
+            },
+            placeholder: {
+                type: String,
+                default: 'Start typing to search...'
+            },
+        },
+        data() {
+            return {
+                filteredItems: undefined,
+                query: '',
+                itemsWithIds: [],
+                results: [],
+                focused: false,
+                warnedAbout: [] // prevents multiple identical warnings
+            }
+        },
+        watch: {
+            items: function() {
+                if (this.items) {
+                    this.itemsWithIds = this.assignItemsUIDs(this.items)
+                }
+
+            }
+        },
+        mounted() {
+            this.reset()
+        },
+        methods: {
+            reset() {
+                this.query = ''
```

Comment:
Should this also reset `filteredItems` and a the others?

Reply:
i think this method should be removed altogether - its not being used internally.
Not sure if we'd want to have it available for external use